### ISSUE TITLE
Orientation change doesn't trigger size change

### ISSUE THREAD

**Issue by lucadegasperi** [aFarkas/lazysizes#121]
Hello, on polyfilled browsers like safari and safari for iOS when the device is rotated, a new width isn't set on the image. Is there a way to have the image always at 100% of the container instead of a pixel value?

**Comment by aFarkas** [comment]
I don't really understand, what you mean. In case of an orientation change the ``sizes`` attribute is also changed, right?

``%`` units are not allowed for ``sizes``, but you can use ``style="width: 100%;"``.

**Comment by lucadegasperi** [comment]
No it is not, the image remains wider than the container in case of a scale down and smaller in case of a scale up on orientation change. The script applies an inline style to the img element which doesn't change when the orientation changes.
```<img src="low res img" data-sizes="auto" data-srcset="all the different srcs" class="lazyload">```

**Comment by aFarkas** [comment]
lazySizes does clearly not add an inline styles. While it is really hard to understand what is going wrong without a simplified testcase. (Hint: It would help me a lot to have one)

The hole point of the `` data-sizes="auto"``  feature is to transform your current CSS width information into the ``sizes`` attribute.

From what you are saying it sounds, that you actually do not style the width/dimensions of your images. So here is the question: Do you have any explicit width styles applied to your images? for example ``img {width: 100%}``.

In case you want to customize the lazySizes ``width``  calculation, this can be done with the following code:

```js
document.addEventListener('lazybeforesizes', function(e){
    e.detail.width = yourStyleCalculation(e.target, e.detail.width);
});
```

**Comment by lucadegasperi** [comment]
I'm using lazySizes together with picturefill. Images only have a ```max-width: 100%``` applied to them via CSS. no other values. what I get in the inspector is the image gets a ```width="300"``` attribute set to it, the same value as the ```sizes``` attribute.

**Comment by aFarkas** [comment]
The ``width="300"`` is coming from picturefill. In case you do not specify the ``width`` of your images, it is not possible to calculate the width in a generic way. From my viewpoint it is even better to also [specify the height](https://github.com/aFarkas/lazysizes/tree/master#user-content-specify-dimensions).

In case you can define a rule for calculating the right ``sizes``  you can do so by using the ``lazybeforesizes`` event. For example:

```js
document.addEventListener('lazybeforesizes', function(e){
    //take computed width of image or its container, whatever is larger
    //but be careful this can waste data, especially with portrait images
    e.detail.width = Math.max(e.target.parentNode.offsetWidth, e.detail.width);
});
```

Basically this should help you. If don't please provide a testcase, which shows your concrete problem to solve otherwise please close.

**Comment by lucadegasperi** [comment]
Weird because I have used picturefill until now (not in conjunction with lazysizes) and it never setted any width on the img element. I have fluid width containers and the images in the container need to resize as the container (and the page) resize, right now they don't because somewhere a width gets set on the image.

**Comment by aFarkas** [comment]
This is a feature of pf 2.3. You can override it by using ``img {width: auto;}``. In general you should have a similar behavior in native browsers without seeing a ``width`` atttribute.

**Comment by lucadegasperi** [comment]
added ```img {width: auto;}``` but doesn't look like to solve the problem on polyfilled browsers.

**Comment by aFarkas** [comment]
@lucadegasperi 
I really can't help you, if I don't have a testcase with your concrete problem.

In general this is how ``data-sizes="auto"``, it takes the ``offsetWidth`` of the image and adds it to the ``sizes`` attribute. In case this width is smaller than 40pixel it traverses up the dom tree until it finds a width that is larger than 40pixel.

So if you don't specify the width of your image, it will initially traverse at least one element up to get the on resize on the other hand the image will have a width above 40pixel and in this case the algorithm will take this. In most cases this width is static until it hits the ``max-width`` bound.

This is by design, because you need to specify somehow the width. In case you want something different you can extend the sizes calculation with the ``lazybeforesizes`` event.

In case you don't know how, I can only help you if I get a testcase with your exact problem.

**Comment by lucadegasperi** [comment]
This solved the problem: ```img[data-sizes="auto"] { display: block; width: 100%; }```
